book/scale_nodes.rst

```diff
@@ -0,0 +1,360 @@
+ORO Application Scaling - Set-up Instructions
+=============================================
+
+Introduction:
+-------------
+
+Slow-down of the server response when application is under heavy loads may result in deterioration of the 
+user-experience and decrease of the user-satisfaction level. Therefore, we have to somehow optimize and speed-up the 
+response time.
+
+The simplest solution is to scale-up web nodes of the application. For example, scaling of just two web nodes can 
+improve the response time twice (at certain amount of request). The article describes how to scale the web-nodes for the 
+application.
+
+Preliminary Recommendations:
+----------------------------
+
+- Share the *[Application root]/app/cache* folder among all the web nodes, for example using NFS.
+
+- Use Nginx (http://nginx.org/) as a proxy balancer
+
+- Use REDIS (http://redis.io/) as a data cache for Annotations, Metadata, ACL, etc.
+
+- Share *[Application root]/app/attachment* folder among all the web nodes 
+
+.. hint::
+
+    The ways to share *[Application root]/app/attachment* are described in the Attachments section hereinafter. 
+
+Additional Requirements to the Bundles. Known Issues and Solutions:
+-------------------------------------------------------------------
+
+- **Maintenance mode**: there shouldn't be any problems with operations that require the maintenance mode, as default 
+  configuration settings place the 'maintenance_lock' file in the cache directory, which is already shared among the 
+  web-nodes.
+  If you want to store the lock file in another location, see the `documentation on the 
+  LexicMaintenanceBundle <https://github.com/lexik/LexikMaintenanceBundle/blob/master/Resources/doc/index.md>`_.
+
+- **CRON** jobs: there shouldn't be any problems, as the queue is stored in the Database, so all the web-nodes will 
+  know the actual state of each job. 
+  By default the job daemon is configured to run 5 concurrent (simultaneous) jobs. So, for example, if we have 5 web 
+  farm nodes we can decrease this value to 1. In this case, each node will run only 1 CRON job. In some cases it can 
+  also be more efficient to have an independent node just for CRON jobs.
+
+- **REDIS**: use `SncRedisBundle <https://github.com/snc/SncRedisBundle>`_
+  
+  - was tested with: the version 1.1.9 
+
+  - was tested with PhpRedis extension, however you may also use Predis (but only versions 0.8.* are supported by 
+    the SncRedisBundle).
+
+    - The PhpRedis extension file (.so) can be installed via the package manager
+
+      - for Linux: yum/dnf/apt-get install php-pecl-redis
```

Comment:
Just a heads up, in ubuntu 14.04 and php 5.6 the right command is: 
`apt-get install php5-redis`

otherwise:
`Unable to locate package php-pecl-redis`

I would mention about editing this in php.ini
```
session.save_handler = redis
session.save_path = 'tcp://localhost:6379'
```